src/main/java/com/jit/studentmanagementsystem/controllers/TeacherController.java

```diff
@@ -0,0 +1,55 @@
+package com.jit.studentmanagementsystem.controllers;
+
+import com.jit.studentmanagementsystem.models.Teacher;
+import com.jit.studentmanagementsystem.repositories.TeacherRepository;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/teachers")
+public class TeacherController {
+    @Autowired
+    private TeacherRepository teacherRepository;
+
+    @GetMapping
+    public List<Teacher> list() {
+        return teacherRepository.findAll();
+    }
+
+    @GetMapping("{id}")
+    public Teacher get(@PathVariable Long id) {
+        return teacherRepository.getOne(id);
+    }
+
+    @PostMapping
+    public Teacher create(@RequestBody final Teacher teacher) {
+        return teacherRepository.saveAndFlush(teacher);
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.DELETE)
+    public void delete(@PathVariable Long id) {
+        teacherRepository.deleteById(id);
```

Comment:
delete (and update) can return count of changed rows, it is sometimes useful to double check with that

Reply:
I was supposing that delete and update here applies to just one row.

Reply:
but it may change 0 rows if database is empty

Reply:
it is much more useful in more complex cases

Reply:
ok, I understand now. I will have to make a count on the database rows before and after deleting the row.

Reply:
not necessary, modifying statement will return that changed rows count